### PR TITLE
Bugfix api issue

### PR DESCRIPTION
**Pull request by ashu555**
Fixes the security issue when no valid plan exists for a product.

### CODE REVIEW

File: app/Http/Controllers/Front/BaseCartController.php [hunk 1]
```diff
@@ -237,14 +243,12 @@ private function getCartValues($productId, $canReduceAgent = false)
...
-            } else {
-                $agtqty = $agtqty * 2;
-                $price = \Cart::getTotal() * 2;
-            }
+        if ($canReduceAgent) {
+            $agtqty = $agtqty / 2;
+            $price = \Cart::getTotal() / 2;
+        } else {
+            $agtqty = $agtqty * 2;
+            $price = \Cart::getTotal() * 2;
```
Comment (mverma16) [inline app/Http/Controllers/Front/BaseCartController.php:251]: write doc comment to explain this logic

**Comment by bhanu2217** [comment]
@ashu555 What is the status on this?

File: app/Http/Controllers/Front/BaseCartController.php (hunk 1 above)
Comment (ashu555) [reply app/Http/Controllers/Front/BaseCartController.php:251]: done

File: app/Plugins/Stripe/views/middle-page.blade.php [hunk 2]
```diff
@@ -96,7 +96,7 @@
                         </div>
                         <div class="form-group row">
                             <div class="col-md-12">
-                                <input id="amount" type="text" value={{Session::get('amount')}} class="form-control @error('amount') is-invalid @enderror" required autocomplete="current-password" name="amount" placeholder="Amount" readonly>
+                                <input id="amount" type="text" value={{$total}} class="form-control @error('amount') is-invalid @enderror" required autocomplete="current-password" name="amount" placeholder="Amount" readonly>
```
[the remark this answers is not in the document]
Comment (ashu555) [reply app/Plugins/Stripe/views/middle-page.blade.php:99]: checked

File: app/Http/Controllers/Front/BaseCartController.php [hunk 3]
```diff
@@ -212,22 +215,35 @@ public function updateAgentQty(Request $request)
...
 
             return successResponse('Cart updated successfully');
         } catch (\Exception $ex) {
             return errorResponse($ex->getMessage());
         }
     }
 
+    /**
+     * The method returns the updated price, no of agents and currency of the product added to the cart
+     * Since this method is called when the the user has permission to modify agents(set from the admin panel), if the parameter $canReduceAgent is true the agent quantity and the cart total will be divided by two else they will be multiplied by two.
```
Comment (mverma16) [inline app/Http/Controllers/Front/BaseCartController.php:235]: reason for multiplication or divide by 2?

File: app/Http/Controllers/Front/BaseCartController.php (hunk 1 above)
Comment (mverma16) [reply app/Http/Controllers/Front/BaseCartController.php:261]: I meant the document the logic definition why are we dividing or multiplying by 2?

File: config/app.php [hunk 4]
```diff
@@ -100,7 +100,8 @@
     |--------------------------------------------------------------------------
     |
     | This key is used by the Illuminate encrypter service and should be set
-//
+
+'App\Plugins\Stripe\ServiceProvider',
 'App\Plugins\stripe\ServiceProvider',
 'App\Plugins\stripr\ServiceProvider',//
 'App\Plugins\Paypal\ServiceProvider',//
```
Comment (mverma16) [inline config/app.php:107]: this is not the correct place to register ServiceProviders

File: app/Http/Controllers/Front/BaseCartController.php (hunk 3 above)
Comment (ashu555) [reply app/Http/Controllers/Front/BaseCartController.php:235]: updated

File: config/app.php (hunk 4 above)
Comment (ashu555) [reply config/app.php]: Those were unnecessary few lines added. Removed

**Comment by bindu95** [comment]
> Please check for the other case reported by Avinash, make sure to not leave any stone unturned.

@ashu555 we fixed all the issues reported by @avinash403 sir in this pr?

**Comment by Redenkaiah** [comment]
**Scenarios:**
For Agent: o Checked
                 o Unchecked
For Product quantity: o Checked
                                   o Unchecked
**Checking URLs**: For agents checked -- http://localhost/agora-invoicing-community/public/update-agent-qty?productid=14
                                                             http://localhost/agora-invoicing-community/public/reduce-agent-qty?productid=14
                     For agents Unchecked -- http://localhost/agora-invoicing-community/public/update-agent-qty?productid=14
                                                             http://localhost/agora-invoicing-community/public/reduce-agent-qty?productid=14

   For Product quantity checked--http://localhost/agora-invoicing-community/public/update-qty?productid=14
                                                  http://localhost/agora-invoicing-community/public/reduce-product-qty?productid=14
 For product quantity Unchecked--http://localhost/agora-invoicing-community/public/update-qty?productid=14
                                                  http://localhost/agora-invoicing-community/public/reduce-product-qty?productid=14
Screenshot_163
**OuttPuts::** First hit the URL-Showing method not allowed
                Once edit and resend POST method and pass csrf token---
                       Some places 200 response is coming-updated values
                       Some places 302 response is coming-not updated values 
                       Some places  Oh snap cannot modify quantity is coming
Screenshot_165
Screenshot_164
Screenshot_162
This is last commit:
Screenshot_166

**Pull request merged**